### PR TITLE
Support SD2.1 and SD2.1-v. Support auto-download ckpt. Arg default changed: wd1e-6, dpm++, bs4

### PR DESCRIPTION
**Pull request by SamitHuang**
- Add SD version control
- Default SD version for text to image is now 2.1
- Default `weight_decay` for training or fine-tuning is reverted back to 1e-6, due to the generated images by LoRA using this value looks better than 1e-2 
- Default `n_samples` (i.e. batch size) (for text_to_image) changed to 4, to make 768x768 generation runnable in default setting.
- Default `sampler` changed to dpm_solver_pp, and `sampling_steps` changed to 20

Now it is able to auto download the required checkpoint for inference for different versions!

### CODE REVIEW

**Comment by wtomin** [comment]
[stable-diffusion-2-1](https://huggingface.co/stabilityai/stable-diffusion-2-1) official reference contains two models: non-ema and ema checkpoints. It needs to be specified which one is converted into MindSpore model weights.

File: examples/stable_diffusion_v2/text_to_image.py [hunk 1]
```diff
@@ -433,27 +451,34 @@ def main(args):
...
     if args.config is None:
-        if args.version in ["1.5_cn", "wukong"]:
-            args.config = "configs/v1-inference-chinese.yaml"
-        elif args.version.startswith("1."):  # 1.x, 1.5
-            args.config = "configs/v1-inference.yaml"
-        elif args.version == "2.0-v768":
-            args.ckpt_path = "models/v2-vpred-inference.yaml"
-        else:
-            args.config = "configs/v2-inference.yaml"
+        args.config = "configs/" + _version_cfg[args.version][1]
```
Comment (DongyaoZhu) [inline examples/stable_diffusion_v2/text_to_image.py:481]: os.path.join?

File: examples/stable_diffusion_v2/README.md [hunk 2]
```diff
@@ -61,13 +61,16 @@ Currently, we provide pre-trained stable diffusion model weights that are compat
...
 | 2.0-inpaint      | image inpainting | [sd_v2_inpaint-f694d5cf.ckpt](https://download.mindspore.cn/toolkits/mindone/stable_diffusion/sd_v2_inpaint-f694d5cf.ckpt) | [stable-diffusion-2-inpainting](https://huggingface.co/stabilityai/stable-diffusion-2-inpainting) | 512x512|
 | 1.5       | text-to-image | [sd_v1.5-d0ab7146.ckpt](https://download.mindspore.cn/toolkits/mindone/stable_diffusion/sd_v1.5-d0ab7146.ckpt) | [stable-diffusion-v1-5](https://huggingface.co/runwayml/stable-diffusion-v1-5) | 512x512 |
 | wukong    | text-to-image |  [wukong-huahua-ms.ckpt](https://download.mindspore.cn/toolkits/minddiffusion/wukong-huahua/wukong-huahua-ms.ckpt) |  | 512x512 |
 | wukong-inpaint    | image |  [wukong-huahua-inpaint-ms.ckpt](https://download.mindspore.cn/toolkits/minddiffusion/wukong-huahua/wukong-huahua-inpaint-ms.ckpt) |  | 512x512 |
 
+> Resolution refers to the image resolution used in training and is also the optimal choice for image generation. Other vaules (divisable by 64) are worakable but may degrade the generation quality.
```
Comment (DongyaoZhu) [inline examples/stable_diffusion_v2/README.md:73]: worakable -> usable; lead to a degrade in .. (gpt)

**Review by DongyaoZhu — APPROVED** [review]
weak accept; minor revision

File: examples/stable_diffusion_v2/README.md (hunk 2 above)
Comment (SamitHuang) [reply examples/stable_diffusion_v2/README.md]: fixed

File: examples/stable_diffusion_v2/text_to_image.py (hunk 1 above)
Comment (SamitHuang) [reply examples/stable_diffusion_v2/text_to_image.py:481]: same. the folder name is fixed anyway.

Comment (DongyaoZhu) [reply examples/stable_diffusion_v2/text_to_image.py:481]: I think it's a backward slash \ on windows

Comment (SamitHuang) [reply examples/stable_diffusion_v2/text_to_image.py:481]: agree. fixed

**Pull request merged**